src/clips-specs/rcll-central/execution-monitoring.clp

```diff
@@ -1230,3 +1230,26 @@
   =>
   (modify ?gm (restricted-to nil))
 )
+
+(defrule execution-monitoring-change-robot-pos-preemptively
+" Update the robot position if it is actually not at the current position anymore.
+"
+  (declare (salience ?*MONITORING-SALIENCE*))
+  (goal (id ?goal-id))
+  (goal-meta (goal-id ?goal-id) (assigned-to ?robot))
+
+  ?pa <- (plan-action (plan-id ?plan-id) (goal-id ?goal-id)
+    (id ?id) (state RUNNING)
+    (action-name move|go-wait) (param-values ?robot ?mps ?side $?to)
+  )
+  ?at <- (domain-fact (name at) (param-values ?robot ?mps ?side))
+  (domain-fact (name zone-content) (param-values ?zone ?mps))
+
+  (Position3DInterface (id ?if-id&:(eq ?if-id (remote-if-id ?robot "Pose"))) (translation ?x ?y ?z))
```

Comment:
This doesn't work in simulation (atm), right?

Reply:
correct, since the navgraph and it's position faker is gone now, it does not work anymore.